### PR TITLE
feat(l1): eth capability transactions message

### LINKED ISSUE #378: Transactions (0x02)

### PR DESCRIPTION
**Pull request by fkrause98**
**Motivation**

- Support the eth capability message 'transactions' which broadcasts transactions.
**Description**
- Implement 'transactions' message handling.
- Use a Tokio broadcast channel to communicate different connections.
- Re-write transaction message encoding+decoding.
- Opened issue to add the received transactions to the mempool.

Closes #378.

### CODE REVIEW

File: crates/networking/p2p/net.rs
Comment (ElFantasma) [inline crates/networking/p2p/net.rs:21]: nit: the `X` should be lowercase: `RLPxMessage`

File: crates/networking/p2p/net.rs
Comment (ElFantasma) [inline crates/networking/p2p/net.rs:95]: Maybe it's time to add a `RLPxConnContext` struct to put together the table, signer, storage and channel(s) in a single parameter.

(If not here, maybe open a ticket to deal with it)

File: crates/networking/p2p/rlpx/connection.rs
Comment (ElFantasma) [inline crates/networking/p2p/rlpx/connection.rs:198]: Nice!

Will have conflicts with my PR, though 😬

File: crates/networking/p2p/rlpx/connection.rs
Comment (ElFantasma) [inline crates/networking/p2p/rlpx/connection.rs:208]: In my [PR](https://github.com/lambdaclass/lambda_ethereum_rust/pull/1136) I'm simplifying this function to extract the message handling in another function. 
That will be better for this update, I can merge this into mine once it get merged

File: crates/networking/p2p/rlpx/connection.rs
Comment (ElFantasma) [inline crates/networking/p2p/rlpx/connection.rs:218]: This `match` will grow as we will handle all message types here. In that sense maybe we should keep it simple: just call another layer function and send a response if needed.

This could look something like:
```rust
txs_msg @ Message::Transactions(_) => {
   self.broadcast(txs_msg)?
}
```

File: crates/networking/p2p/rlpx/connection.rs
Comment (ElFantasma) [inline crates/networking/p2p/rlpx/connection.rs:206]: While I see the goal here, there is still a problem (I think): Status (as well as Ping, Pong and Disconnect) messages can still be received (and acted upon) inside the loop.

Comment (ElFantasma) [reply crates/networking/p2p/rlpx/connection.rs:206]: Maybe we need a new State (eg. `StatusStablished`) or a flag to determine if an `eth` capability protocol was stablished, and do not brodcast transactions if that's not the case.

File: crates/networking/p2p/rlpx/connection.rs
Comment (ElFantasma) [inline crates/networking/p2p/rlpx/connection.rs:464]: I'm not so sure about this.

Disconnect, Ping and Pong should be in the main loop. (and probably Status too). Also if Disconnect is received, it makes no sense to wait for a Status)

File: crates/networking/p2p/rlpx/error.rs
Comment (ElFantasma) [inline crates/networking/p2p/rlpx/error.rs:35]: `BroadcastError` or `BroadcastFailed`?

**Comment by ElFantasma** [comment]
It looks good on general level, but have some concerns on the handling of Status messages. It can even be the case that `eth` is not supported and the main loop should still be running for the other capabilities. Let's talk if you want to discuss it.

File: crates/networking/p2p/net.rs
Comment (fkrause98) [reply crates/networking/p2p/net.rs:95]: Opened an issue [here](https://github.com/lambdaclass/lambda_ethereum_rust/issues/1140) for it.


Maybe I'll add it naturally on future P2P PRs if I see it's becoming hard to handle with arguments alone.

File: crates/networking/p2p/rlpx/connection.rs
Comment (fkrause98) [reply crates/networking/p2p/rlpx/connection.rs:218]: Done on this [commit](https://github.com/lambdaclass/lambda_ethereum_rust/pull/1118/commits/96b738976fab9bfb85abaccc2f6a7f29fd053514#)

File: crates/networking/p2p/rlpx/error.rs
Comment (fkrause98) [reply crates/networking/p2p/rlpx/error.rs]: Renamed to BroadcastError

**Review by ElFantasma — APPROVED** [review]
Nice!
There is still some room for improvement on Status reception, but since I'm working on Status validations (#420) I think we can merge this and correct any problem on that PR

**Pull request merged**
